### ISSUE TITLE
Fail to run static pod without master running

### ISSUE THREAD

**Issue by liyimeng** [k3s-io/k3s#10036]
**Environmental Info:**
K3s Version: 

1.28.6

Node(s) CPU architecture, OS, and Version: 

4 core arm64
Cluster Configuration:

1 master + 1 worker
**Describe the bug:**

I have a static pod running in each node, both nodes went into failure for a power lost.  For some reason, the master node was not able to recovery, but worker node boot back as normal. However, k3s service stuck at 

```
time="2024-04-28T02:53:14Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33176->127.0.0.1:6444: read: connection reset by peer"
```  
hence my static pod was not able to come back. 

I am wondering if this is an expected behaviour. Could it will be nice to have containerd and kubelet up and run, then static pod start. Let the kubelet try to connecting  back when master is available later on?
 

**Steps To Reproduce:**

- Installed K3s on two node, one run as master, another run as agent.
- Deploy static pod yaml in /var/lib/rancher/k3s/server/manifests
- Static pod up and run in both nodes
- Power off both nodes
- Power on work node, but let master off
- k3s service fails to start, so the static pod
**Expected behavior:**

 k3s service on work node  start containerd, kubelet, and static pod,  keep trying to reach master until master node is online. 

**Actual behavior:**

k3s  service on work node failed to start

**Additional context / logs:**

```
time="2024-04-28T02:52:00Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33126->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:06Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33130->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:12Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33136->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:18Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33140->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:25Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33144->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:31Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33148->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:37Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33152->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:43Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33156->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:49Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33160->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:52:55Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33164->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:53:01Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33168->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:53:08Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33172->127.0.0.1:6444: read: connection reset by peer"
time="2024-04-28T02:53:14Z" level=error msg="failed to get CA certs: Get \"https://127.0.0.1:6444/cacerts\": read tcp 127.0.0.1:33176->127.0.0.1:6444: read: connection reset by peer"

```

**Comment by liyimeng** [comment]
I see we have [allow this](https://github.com/k3s-io/k3s/blob/94e29e2ef5d79904f730e2024c8d1682b901b2d5/pkg/agent/run.go#L478) for etcdagent node, can it be extended to normal agent node?

**Comment by brandond** [comment]
This is a duplicate of https://github.com/k3s-io/k3s/issues/1686

K3s agents do not start the container runtime or kubelet until a server is reachable to provide up-to-date certificates and configuration. Because the kubelet isn't started yet, there is no issue with Kubernetes conformance - conformance places no requirements on how a distribution operates prior to startup of Kubernetes itself, or what startup dependencies a distribution enforces.

Although there have been discussions on this topic in the past, we are not planning to relax this requirement at this time
